src/bin/pg_autoctl/state.c

```diff
@@ -492,29 +494,32 @@ NodeStateFromString(const char *str)
 /*
  * epoch_to_string converts a number of seconds from epoch into a date time
  * string.
+ *
+ * This string is stored in buffer and for ease of use of the function this
+ * buffer is also returned. The buffer should (at least) be MAXCTIMESIZE large.
  */
 const char *
-epoch_to_string(uint64_t seconds)
+epoch_to_string(uint64_t seconds, char *buffer)
 {
-	if (seconds > 0)
+	if (seconds <= 0) {
+		strlcpy(buffer, "0", MAXCTIMESIZE);
+		return buffer;
+	}
+	/*
+	 * ctime_r returns a string that ends with \n, which we don't want.
+	 * Replace it with a null string terminator, even if that means we lose
+	 * a precious byte of memory here..
+	 */
+	ctime_r((time_t *) &seconds, buffer);
```

Comment:
When discarding the return value of a function call, we often write `(void) funcall(...)` to make it obvious to the readers: we are discarding it on purpose.

Reply:
I'm actually using the result now, since it indicates if it errors.